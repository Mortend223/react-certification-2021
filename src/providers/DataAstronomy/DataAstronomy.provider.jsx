import React, { createContext, useContext, useEffect, useReducer } from 'react';
import PropTypes from 'prop-types';

const DataAstromyContext = createContext(null);

function DataAstronomyReducer(state, action) {
  switch (action.type) {
    case 'updateDate': {
      return { ...state, searchDate: action.date };
    }
    default: {
      throw new Error(`Unhandled action type: ${action.type}`);
    }
  }
}

function DataAstronomyProvider({ children }) {
  const [state, dispatch] = useReducer(DataAstronomyReducer, {
    searchDate: new Date(),
  });

  const onChangeDate = (value) => {
    dispatch({ type: 'updateDate', date: value });
  };

  useEffect(() => {}, [state.searchDate]);

  return (
    <DataAstromyContext.Provider
      value={{
        searchDate: state.searchDate,
        onChangeDate,
      }}
    >
      {children}
    </DataAstromyContext.Provider>
  );
}

function useDataAstronomy() {
  const context = useContext(DataAstromyContext);
  if (context === undefined) {
    throw new Error('useDataAstronomy must be used within a DataAstronomyProvider');
  }
  return context;
}

DataAstronomyProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export { useDataAstronomy };
export default DataAstronomyProvider;
