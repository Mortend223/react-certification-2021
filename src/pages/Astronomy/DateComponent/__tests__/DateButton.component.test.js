import React from 'react';
import { render } from '@testing-library/react';

// Components
import DateButtonComponent from '../DateButton.component';

// Hooks
import { useDataAstronomy } from '../../../../providers/DataAstronomy/DataAstronomy.provider';

jest.mock('../../../../providers/DataAstronomy/DataAstronomy.provider');

describe('DateButton Component', () => {
  it('Should show DateButton Component', () => {
    useDataAstronomy.mockReturnValue({
      searchDate: '2021-11-21',
      onChangeDate: jest.fn(),
    });
    const error = {
      code: 400,
      msg: 'Some text',
    };
    const container = render(<DateButtonComponent errorResponse={error} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
