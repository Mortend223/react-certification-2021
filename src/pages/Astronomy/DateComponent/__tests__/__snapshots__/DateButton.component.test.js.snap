// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateButton Component Should show DateButton Component 1`] = `undefined`;
